### PR TITLE
software: activation - fix stop button behavior

### PR DESCRIPTION
**Pull request by csommelet**
There is a small risk that the millis() counter makes a round trip and test `(millis() - m_timeOfLastStopPushed) < SECOND_STOP_MAX_DELAY_MS` becomes true again unexpectedly

### CODE REVIEW

File: src/software/firmware/srcs/activation.cpp [hunk 1]
```diff
@@ -47,3 +47,13 @@ void ActivationController::onStopButton() {
...
+
+void ActivationController::refreshState() {
+    // If the 2nd STOP deadline is exceeded, switch back
+    // to running state. This is needed to make sure millis() counter
+    // overflow does not make the test invalid.
+    if ((m_state == RUNNING_READY_TO_STOP)
+        && ((millis() - m_timeOfLastStopPushed) >= SECOND_STOP_MAX_DELAY_MS)) {
+        m_state = RUNNING;
+    }
+}
```
Comment (jabby) [inline src/software/firmware/srcs/activation.cpp:59]: We need an empty line at the end of the file.
If you don't have time, we can take care of this point. 

PS: Thanks for all your PRs and reviews.

Comment (csommelet) [reply src/software/firmware/srcs/activation.cpp:59]: I am on it. Still waiting for my push to finish

Comment (csommelet) [reply src/software/firmware/srcs/activation.cpp:59]: I dont know why I cannot push/fetch my remote. I let you make the change then.

File: src/software/firmware/srcs/activation.cpp [hunk 2]
```diff
@@ -47,3 +47,13 @@ void ActivationController::onStopButton() {
         break;
     }
 }
+
+void ActivationController::refreshState() {
+    // If the 2nd STOP deadline is exceeded, switch back
+    // to running state. This is needed to make sure millis() counter
+    // overflow does not make the test invalid.
```
Comment (dsferruzza) [inline src/software/firmware/srcs/activation.cpp:54]:
```suggestion
    // overflow does not make the test invalid
```

**Pull request merged**